### PR TITLE
Implement postprocessing electron

### LINKED ISSUE #1545: Convert post-processing to an electron
Remove postprocessing from the [runner.](https://github.com/AgnostiqHQ/covalent/blob/os-enhancements-preview/covalent_dispatcher/_core/runner.py) **All electrons look the same to the runner — no more `workflow_executor`**
Also, the SDK should bind the lattice’s `workflow_executor` to each sublattice/postprocessing electron’s `executor` during `build_graph()`.

### PR DESCRIPTION
**Pull request by FyzHsn**
Fixes #1545 

- [ ] I have added the tests to cover my changes.
- [ ] I have updated the documentation and CHANGELOG accordingly.
- [ ] I have read the CONTRIBUTING document.

### CODE REVIEW

File: covalent/_workflow/lattice.py [hunk 1]
```diff
@@ -87,6 +88,9 @@ def __init__(
 
         self.workflow_function = TransportableObject.make_transportable(self.workflow_function)
 
+        # Bound electrons are defined as electrons with a valid node_id, since it means they are bound to a TransportGraph.
```
Comment (kessler-frost) [inline covalent/_workflow/lattice.py:91]: I see, we have a definition here as well.

File: covalent/_workflow/lattice.py [hunk 2]
```diff
@@ -237,12 +239,22 @@ def build_graph(self, *args, **kwargs) -> None:
...
                 except Exception:
                     warnings.warn(
                         "Please make sure you are not manipulating an object inside the lattice."
                     )
                     raise
 
+        from .postprocessing import Postprocessor
+
+        pp = Postprocessor(lattice=self)
+
```
Comment (kessler-frost) [inline covalent/_workflow/lattice.py:252]: Can you put this inside the if condition if it is only valid when the condition is true?

Comment (kessler-frost) [reply covalent/_workflow/lattice.py:252]: Also, if this import is not turning out to be cyclical then maybe we can put it on the top of the file.

File: covalent/_workflow/postprocessing.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+log_stack_info = logger.log_stack_info
+
+
+class Postprocessor:
+    def __init__(self, lattice) -> None:
+        self.lattice = lattice
+
+    def _is_postprocessable_node(self, tg: _TransportGraph, node_id: int) -> bool:
+        """Filter nodes that should be included in postprocessing."""
+        node_name = tg.get_node_value(node_id, "name")
```
Comment (kessler-frost) [inline covalent/_workflow/postprocessing.py:52]: Missing args and return value in docstring here.

File: covalent/_workflow/postprocessing.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        """
+        Post processing function to be called after the lattice execution.
+        This takes care of executing statements that were not an electron
+        but were inside the lattice's function. It also replaces any calls
+        to an electron with the result of that electron execution, hence
+        preventing a local execution of electron's function.
+        Note: Here `node_outputs` is used instead of `electron_outputs`
+        since an electron can be called multiple times with possibly different
+        arguments, but every time it's called, it will be executed as a separate node.
+        Thus, output of every node is used.
```
Comment (kessler-frost) [inline covalent/_workflow/postprocessing.py:83]: Great explanation!

File: covalent/_workflow/postprocessing.py [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+            self.lattice.post_processing = False
+            return result
+
+    def _get_node_ids_from_retval(self, retval: Union["Electron", List, Dict]) -> List[int]:
+        """Get the list of electron node ids from the return value of a lattice function.
+
+        Note: 'retval' here corresponds to the return value of a lattice function.
+
+        Args:
+            retval (_type_): _description_
```
Comment (kessler-frost) [inline covalent/_workflow/postprocessing.py:110]: Missing description.

File: covalent_dispatcher/_db/update.py [hunk 6]
```diff
@@ -87,7 +88,10 @@ def _node(
 
     Returns:
         None
+
     """
+    if not node_name:
```
Comment (kessler-frost) [inline covalent_dispatcher/_db/update.py:93]: If comparing with `None` my suggestion for the ideal way would be to do `if node_name is None`.

**Review by kessler-frost — CHANGES_REQUESTED** [review]
Minor changes, otherwise looks good 👍

File: covalent/_workflow/postprocessing.py (hunk 4 above)
Comment (FyzHsn) [reply covalent/_workflow/postprocessing.py:92]: i think it was Casey's :D

**Review by kessler-frost — APPROVED** [review]

File: covalent/_workflow/postprocessing.py (hunk 4 above)
Comment (cjao) [reply covalent/_workflow/postprocessing.py:92]: Actually it was copied from `runner.py` and formerly `execution.py` -- so probably @kessler-frost 's.

Comment (FyzHsn) [reply covalent/_workflow/postprocessing.py:92]: 😂

**Pull request merged**